rules/do-you-optimize-tinacms-project/rule.md

```diff
@@ -0,0 +1,103 @@
+---
+type: rule
+title: Do you optimize your TinaCMS project for clarity, performance, and reliable builds?
+seoDescription: Learn effective strategies to structure and optimize your
+  TinaCMS project for clarity, performance, and successful builds.
+uri: do-you-optimize-tinacms-project
+authors:
+  - title: "Gilles Pothieu"
+    url: https://www.ssw.com.au/people/gilles-pothieu/
+created: 2024-08-28T16:15:00.000Z
+guid: 3c59f56f-34b0-47a7-902a-718eb340d2e5
+---
+Structuring and optimizing your TinaCMS project is essential to achieve clarity, enhance performance, and prevent build failures. Poorly optimized projects can lead to slow site performance, increased server load, and even failed builds due to excessive or inefficient data requests.
+
+Let’s explore how to structure your project effectively and apply best practices to boost performance both in runtime and during the build process.
+
+<!--endintro-->
+
+## 1. Structuring your TinaCMS Architecture
+
+When working with large datasets or generating multiple subcomponents, following best practices is crucial to maintain performance and clarity.
+
+### ❌ Bad practices
+
+* **Making individual requests with Tina client for each Subcomponent**
+
+  * This method can overwhelm the build process, particularly when dealing with a large number of subcomponents.
```

Comment:
```suggestion
```
this was a point in time issue with Tina and has been fixed

Reply:
✅ Done